packages/core/redux/eos/thunks.js

```diff
@@ -0,0 +1,356 @@
+/**
+ * Copyright 2017–2018, LaborX PTY
+ * Licensed under the AGPL Version 3 license.
+ */
+
+import Eos from 'eosjs'
+import { modalsOpen } from '@chronobank/core/redux/modals/actions'
+import * as EosActions from './actions'
+import * as EosUtils from './utils'
+import Amount from '../../models/Amount'
+import WalletModel from '../../models/wallet/WalletModel'
+import { BLOCKCHAIN_EOS, PAGE_SIZE } from './constants'
+import { describePendingEosTx } from '../../describers'
+import {
+  eosPendingEntrySelector,
+  EOSPendingSelector,
+  EOSSelector,
+  getEosSigner,
+  getEOSTokens,
+  getEOSWallet,
+  getEosWallets,
+} from './selectors/mainSelectors'
+import { ErrorNoticeModel, TransferNoticeModel } from '../../models'
+import { notify } from '../notifier/actions'
+import { getSelectedNetwork } from '../persistAccount/selectors'
+import TxHistoryModel from '../../models/wallet/TxHistoryModel'
+import TokenModel from '../../models/tokens/TokenModel'
+
+const notifyEosTransfer = (entry) => (dispatch) => {
+  const { tx: { from, to, quantity } } = entry
+
+  dispatch(notify(new TransferNoticeModel({ amount: quantity, from, to })))
+}
+
+const notifyEosError = (e) => notify(new ErrorNoticeModel({ message: e.message }))
+
+const eosTxStatus = (key, address, props) => (dispatch, getState) => {
+  const pending = EOSPendingSelector()(getState())
+  const scope = pending[address]
+  if (!scope) {
+    return null
+  }
+  const entry = scope[key]
+  if (!entry) {
+    return null
+  }
+
+  return dispatch(EosActions.eosTxUpdate(
+    key,
+    address,
+    EosUtils.createEosTxEntryModel({
+      ...entry,
+      ...props,
+    }),
+  ))
+}
+
+export const executeEosTransaction = (wallet, quantity, to, memo = '') => async (dispatch) => {
+  const tx = {
+    from: wallet.address,
+    quantity,
+    to,
+    memo,
+  }
+  const entry = EosUtils.createEosTxEntryModel({ tx })
+  await dispatch(EosActions.eosTxCreate(entry))
+  dispatch(submitTransaction(entry))
+}
+
+const processTransaction = ({ entry }) => async (dispatch, getState) => {
+  await dispatch(signTransaction({ entry }))
+  const signedEntry = eosPendingEntrySelector(entry.tx.from, entry.key)(getState())
+  if (!signedEntry) {
+    // eslint-disable-next-line no-console
+    console.error('signedEntry is null', entry)
```

Comment:
`executeEosTransaction` must use try..catch and here we need throw new Error.
Also, in case of any errors `executeEosTransaction` must dispatch `dispatch(eosTxStatus(entry.key, entry.tx.from, { isErrored: true, error }))` or `return notifyEosError('entry is null') // stop execute` (not sure which one is suitable here).